File: lang_cpp/Aksesoris.cpp
#include <iostream> // import library input/output
#include <string> // import library string
#include "Petshop.cpp" // import class Petshop

using namespace std; // menggunakan standard namespace

class Aksesoris : public Petshop{
// membuat class Aksesoris sebagai anak dari class Pershop

    // deklarasi atribut private
    private:
        string jenisAksesoris;
        string bahanAksesoris;
        string warnaAksesoris;


    // deklarasi method public
    public:

        // konstruktor tanpa parameter
        Aksesoris() : Petshop() {
            // Konstruktor default child memanggil konstruktor default parent
        }
        
        // Konstruktor dengan parameter
        Aksesoris(string idProduk, string namaProduk, int hargaProduk, int stokProduk, string jenisAksesoris, string bahanAksesoris, string warnaAksesoris) 
        : Petshop(idProduk, namaProduk, hargaProduk, stokProduk) { // Memanggil konstruktor parent di Initializer List

            // Inisialisasi atribut milik Aksesoris
            this->jenisAksesoris = jenisAksesoris;
            this->bahanAksesoris = bahanAksesoris;
            this->warnaAksesoris = warnaAksesoris;
        }


        // setter untuk atribut jenisAksesoris
        void setJenisAksesoris(string jenisAksesoris){
            this->jenisAksesoris = jenisAksesoris;
        }

        // getter untuk atribut jenisAksesoris
        string getJenisAksesoris(){
            return jenisAksesoris;
        }

        // setter untuk atribut bahanAksesoris
        void setBahanAksesoris(string bahanAksesoris){
            this->bahanAksesoris = bahanAksesoris;
        }

        // getter untuk atribut bahanAksesoris
        string getBahanAksesoris(){
            return bahanAksesoris;
        }

        // setter untuk atribut warnaAksesoris
        void setWarnaAksesoris(string warnaAksesoris){
            this->warnaAksesoris = warnaAksesoris;
        }

        // getter untuk atribut warnaAksesoris
        string getWarnaAksesoris(){
            return warnaAksesoris;
        }

        ~Aksesoris(){
        }

};

File: lang_php/Aksesoris.php
<?php 
include "Petshop.php";
class Aksesoris extends Petshop{
// membuat class Aksesoris sebagai anak dari class Petshop

    // deklarasi atribut class
    private string $jenisAksesoris;
    private string $bahanAksesoris;
    private string $warnaAksesoris;

    public function __construct($idProduk, $namaProduk, $hargaProduk, $stokProduk, $gambarProduk, $jenisAksesoris, $bahanAksesoris, $warnaAksesoris){
        parent::__construct($idProduk, $namaProduk, $hargaProduk, $stokProduk, $gambarProduk);
        $this->jenisAksesoris = $jenisAksesoris;
        $this->bahanAksesoris = $bahanAksesoris;
        $this->warnaAksesoris = $warnaAksesoris;
    }

    // setter untuk atribut jenisAksesoris
    public function setJenisAksesoris($jenisAksesoris){
        $this->jenisAksesoris = $jenisAksesoris;
    }

    // getter untuk atribut jenisAksesoris
    public function getJenisAksesoris(){
        return $this->jenisAksesoris;
    }

    // setter untuk atribut bahanAksesoris
    public function setBahanAksesoris($bahanAksesoris){
        $this->bahanAksesoris = $bahanAksesoris;
    }

    // getter untuk atribut bahanAksesoris
    public function getBahanAksesoris(){
        return $this->bahanAksesoris;
    }

    // setter untuk atribut warnaAksesoris
    public function setWarnaAksesoris($warnaAksesoris){
        $this->warnaAksesoris = $warnaAksesoris;
    }

    // getter untuk atribut warnaAksesoris
    public function getWarnaAksesoris(){
        return $this->warnaAksesoris;
    }

    public function __destruct(){
    }

}
?>